ckanext/cfpb_extrafields/plugin.py

```diff
@@ -75,30 +75,49 @@ def _redirect_to_edit_on_change(self, resource, field):
 
     def _delete_and_rebuild_datadict(self, resource):
         import json
+        import unicodedata
+
         if 'datadict' in resource and 'id' in resource:
-            record = resource['datadict']
-            resource.pop('datadict')
-            json_record = json.loads(record)
+            record = resource.pop('datadict')
+
+            if record:
+                # Cleanse of errant unicode characters
+                record = unicodedata.normalize('NFKD', record).encode('ascii', 'ignore')
```

Comment:
Nice. Does this fix or just patch the problem for release? When ckan reads the data dictionary back from the database, does the formatting remain or does it get stripped out? Do very long entries still get cut off and have some special characters appended?